src/snowflake/snowpark/modin/plugin/extensions/datetime_index.py

```diff
@@ -163,6 +163,7 @@ def __new__(
         index._query_compiler = query_compiler
         # `_parent` keeps track of any Series or DataFrame that this Index is a part of.
         index._parent = None
```

Comment:
how about get a class like IndexParent, and it records the necessary parent information, the constructor can take the df and record the information property, and it can also provide an interface like no_change_on_parent (you can use other name if you have other ones in mind) to help decide wether a replacement should happen, this could help wrap all parent logic at one place and simplify the code handling in the future

Reply:
also, i have a quick question, shouldn't datetime_index an extension of index class, and it can call super().__init__ to init the common part, right? how come we are initialization the same thing at different place?

Reply:
`__init__` in Index.py is used only for docstring generation, the actual initialization is happening in `__new__`. I'm not sure why it was designed this way.
But I created a new class like you suggested to abstract away the common logic.

Reply:
i think the __new__ is implemented because the returned object could be different based on the type. i don't think we have to fix that in this pr, just curious why we didn't call super class for the common part? cc @sfc-gh-azhan do you know?